metabase/metabase
# Allow filtering on all stages in the QB

Epic https://github.com/metabase/metabase/issues/47800
Closes https://github.com/metabase/metabase/issues/47817

Allows to add & remove filters from all query stages in the filter modal, like the notebook does. 

How to verify:
- New -> Question -> Products
- Summarize -> Count, Breakout -> Category (stage 1)
- Summarize -> Count, Breakout -> Category (stage 2)
- Summarize -> Count, Breakout -> Category (stage 3)
- Visualize
- Filter (the filter modal adds stage 4 for convenience)
- Try to add filters on each stage -> Apply filters
- Click Filter again and make sure the filters are in correct groups

<img width="1146" alt="Screenshot 2024-09-10 at 14 48 13" src="https://github.com/user-attachments/assets/a698ba9f-a3f2-4e9e-9230-ec912b7801be">
